File: python/isbn-verifier/isbn_verifier.py
import re
def is_valid(isbn):
    isbn = re.sub("-","", isbn)
    d = list(isbn)
    if len(d) != 10: return False
    if d[9] == 'X': d[9] = 10
    for i, val in enumerate(d):
        try: d[i] = int(val)
        except: return False
    total = d[0] * 10 + d[1] * 9 + d[2] * 8 + d[3] * 7 + d[4] * 6 + d[5] * 5 + d[6] * 4 + d[7] * 3 + d[8] * 2 + d[9] * 1
    return total % 11 == 0

File: swift/isbn-verifier/Sources/IsbnVerifier/IsbnVerifier.swift
class IsbnVerifier {
  static func isValid(_ string: String) -> Bool {
    var d : [Int] = []
    for c in Array(string) {
        if c == "-" {
            continue
        } else if c == "X" {
            d.append(10)
            break
        } else if let num = Int(String(c)) {
            d.append(num)
        } else {
            return false
        }
    }
    var total = 0
    guard d.count == 10 else { return false }
    for (idx, val) in d.enumerated() {
        total += (10 - idx) * val
    }
    return total % 11 == 0
  }
}
